data_collection/gazette/spiders/rj_sao_joao_de_meriti.py

```diff
@@ -0,0 +1,125 @@
+import datetime
+import json
+import re
+
+import scrapy
+from dateutil.rrule import MONTHLY, rrule
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class SpSaoJoaoDeMeritiSpider(BaseGazetteSpider):
+    TERRITORY_ID = "3305109"
+    name = "rj_sao_joao_de_meriti"
+    allowed_domains = ["meriti.rj.gov.br"]
+
+    URL_BEFORE_2018 = "https://meriti.rj.gov.br/dom_mp/"
+    URL_FROM_2018 = (
+        "https://transparencia.meriti.rj.gov.br/diario_oficial_get.php?mesano="
+    )
+
+    start_date = datetime.date(2011, 9, 30)
+    end_date = datetime.date.today()
+
+    # start_date = datetime.date(2017, 12, 25)
+    # end_date = datetime.date(2018, 1, 5)
+
+    def start_requests(self):
+        janeiro_2018 = datetime.date(2018, 1, 1)
+        for month in self.months_interval():
+            if month < janeiro_2018:
+                yield scrapy.Request(
+                    f"{self.URL_BEFORE_2018}{month.year}{month.month:02}.php"
+                )
+            else:
+                yield scrapy.Request(
+                    f"{self.URL_FROM_2018}{month.month}/{month.year}"
+                )
+
+    def parse(self, response):
+        gazettes = []
+
+        if self.URL_BEFORE_2018 in response.url:
+            gazettes = self.parse_month_before_2018(response)
+        else:
+            gazettes = self.parse_month_from_2018(response)

```

Comment:
You do not need this. To remove the entire `parse` method you can define the callback function when you return the request object in the `start_requests`. https://docs.scrapy.org/en/latest/topics/request-response.html?highlight=Request#scrapy.http.Request

`
callback (collections.abc.Callable) – the function that will be called with the response of this request (once it’s downloaded) as its first parameter. For more information see Passing additional data to callback functions below. If a Request doesn’t specify a callback, the spider’s parse() method will be used. Note that if exceptions are raised during processing, errback is called instead.
`

Reply:
It's really better, thank you.